### PR TITLE
Remove encrypted fields from xaction message/issue#427

### PR DESCRIPTION
**Pull request by adecaro**
This is the result of the joint effort of @muralisrini and mine. 

Now, transaction's chaincodeID field is of type bytes. This has allowed us to remove the two fields encrypted\* from the transaction message. The crypto layer has been updated to work with the new structure.

Signed-off-by: Angelo De Caro adc@zurich.ibm.com

### CODE REVIEW

**Comment by muralisrini** [comment]
Thanks, @adecaro. Will check it out.

**Comment by srderson** [comment]
@adecaro There are a number of compile errors in the tests. Can you rebase and resubmit the PR? Also, it would be better if the buildTestTx function in pkg_test.go took a *testing.T param and asserted no error instead of returning the error and making every caller handle that case.

I tried to make a couple of PRs on your branch, but closed them because it was too messy without the rebase.

File: openchain/ledger/blockchain_indexes.go [hunk 1]
```diff
@@ -139,7 +139,14 @@ func getTxExecutingAddress(tx *protos.Transaction) string {
 
 func getAuthorisedAddresses(tx *protos.Transaction) ([]string, *protos.ChaincodeID) {
 	// TODO fetch address from chaincode deployment tx
-	return []string{"address1", "address2"}, tx.GetChaincodeID()
+	// TODO this method should also return error
+	data := tx.ChaincodeID
+	cID := &protos.ChaincodeID{}
+	err := proto.Unmarshal(data, cID)
+	if err != nil {
+		return nil, nil
```
Comment (srderson) [inline openchain/ledger/blockchain_indexes.go]: Should this be logged or should the error be returned?

File: openchain/ledger/pkg_test.go [hunk 2]
```diff
@@ -133,20 +139,24 @@ func (testWrapper *blockchainTestWrapper) populateBlockChainWithSampleData() (bl
 	testWrapper.addNewBlock(block3, []byte("stateHash3"))
 
 	// -----------------------------</Block 3>------------------------------------
-	return allBlocks, allHashes
+	return allBlocks, allHashes, nil
 }
 
-func buildTestTx() (*protos.Transaction, string) {
+func buildTestTx() (*protos.Transaction, string, error) {
```
Comment (srderson) [inline openchain/ledger/pkg_test.go]: Should take *testing.T as a param and AssertNoError instead of returning the error.

**Comment by adecaro** [comment]
Hi @srderson, let me apologize for the issues.
May you check if the conflicts are over now?

**Comment by muralisrini** [comment]
@srderson @adecaro : rebased the ChaincodeID->bytes branch. https://github.com/muralisrini/obc-peer/tree/remove_encrypted_fields_from_xaction_message/issue%23427

**Pull request closed without merging**